firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1733,9 +1733,13 @@ class BrowserViewController: UIViewController,
     /// Call this whenever the page URL changes.
     fileprivate func updateURLBarDisplayURL(_ tab: Tab) {
         guard !isToolbarRefactorEnabled else {
-            let action = ToolbarAction(url: tab.url?.displayURL,
-                                       windowUUID: windowUUID,
-                                       actionType: ToolbarActionType.urlDidChange)
+            let action = ToolbarMiddlewareAction(
+                url: tab.url?.displayURL,
+                traitCollection: traitCollection,
```

Comment:
Minor nit: including `traitCollection` as part of the action feels like it might be a bit of an anti-pattern. It doesn't seem like the trait collection should be part of the associated payload/state, rather we should include a property that describes the meaning of the state we care about (e.g. directly specifying toolbar position etc, as opposed to having the middleware interpret or include logic for checking the trait collection directly). Not sure if that made sense the way I worded it but I wonder if this would be worth hashing out collectively to figure out best practices for this type of thing.